crates/gpio/src/vhu_gpio.rs

```diff
@@ -401,9 +401,14 @@ impl<D: 'static + GpioDevice + Sync + Send> VhostUserBackendMut<VringRwLock, ()>
     }
 
     fn protocol_features(&self) -> VhostUserProtocolFeatures {
-        VhostUserProtocolFeatures::MQ
+        let features = VhostUserProtocolFeatures::MQ
             | VhostUserProtocolFeatures::CONFIG
-            | VhostUserProtocolFeatures::REPLY_ACK
+            | VhostUserProtocolFeatures::REPLY_ACK;
+
+        #[cfg(feature = "xen")]
+        let features = features | VhostUserProtocolFeatures::XEN_MMAP;
```

Comment:
I wonder a bit: The daemons need to special changes to work with Xen, right? Couldn't some higher level of rust-vmm set the feature if xen support ist enabled?

Reply:
There are a couple of things here. First, the XEN_MMAP feature flag is available (for successful build) with or without the `xen` feature. I am confused if I should be adding the above lines under a `#cfg[(feature = "xen")]` or not.

Other than that, the daemons don't need any special changes apart from selecting `xen` features for vhost, etc crates.

And then I didn't understand your last question about higher level rust-vmm stuff.

Reply:
Sorry, let me try to clear it up. I wondered whether `vhost-user-backend` (which already has the xen feature) could add the protocol feature without the vhost-device daemons needing to spell it out themselves? If it is transparent to the daemons, it would probably lead to people not listing the feature and unnecessarily restricting their daemon not to be supported by Xen?

Reply:
How about this:

https://github.com/rust-vmm/vhost/pull/160/commits/e47c7792285bc53e05e5933f455d17650245bea6 

? 

@jiangliu Any suggestions ?